### DISCUSSION TITLE
Confused with radius of search_params in COSINE and HNSW

### DISCUSSION THREAD

**Discussion by UselessXiaoYao** [milvus-io/milvus#35880] [Q&A and General discussion]
I have build a collection with index_param:
`
index_params.add_index(
            field_name="video_embed",
            index_type="HNSW",
            metric_type="COSINE",
            params={"M": 32, "efConstruction": 500},
        )
`
And when I use client.search() the data existed in the collection with search_params:
`
search_params = {
                "metric_type": "COSINE",
                "params": {
                    "radius": **0.1**, # Determines the threshold of least similarity.
                    "range_filter":1.0,
                }
            }
`
I can get the data's self as result, such as:
{"id": "data-self-id", "distance": 0.9999995827674866}

But, if I use search_params:
`
search_params = {
                "metric_type": "COSINE",
                "params": {
                    "radius": **0.9**, # Determines the threshold of least similarity.
                    "range_filter":1.0,
                }
            }
`
**some datas** can not got their-selfs in results. And in this condition, no result returned.

And no range_filter, has the same result.

This is a problem for me now. I can't use search the most similar results to filter existed data.

**Comment by UselessXiaoYao** [comment]
I have tried adjust the radius value, and on some data, I should set radius to 0.57805, if the value more than this, I can not get the data point itself in search result. And this value is not universally applicable.

**Comment by yhmo** [comment]
It seems the "radius" and "range_filter" are not correctly set.
This document tells how to set the two parameters in different situations:
https://milvus.io/docs/single-vector-search.md#Range-search


For IP/COSINE metric, distance/similarity range is [1.0, -1.0], 1.0 is the most similar item.
For L2 metric, distance range is [0.0, +∞], 0.0 is the most similar item.

"radius" is equal to the "range_max", "range_filter" is equal to "range_min".
This graph shows how they work for range filtering:
![range_search drawio](https://github.com/user-attachments/assets/c2023ea6-e229-4a2c-8c7f-1a02d24855d9)



So, for COSINE metric, you should set radius smaller than range_filter :
search_params = { "metric_type": "COSINE", "params": { "radius": **0.9**, # Determines the threshold of least similarity. "range_filter":1.0, } }

**Reply by xiaofan-luan** [reply]
@yhmo 
I like this illustration. can we put this on our document? 
We also need to note that 1.0 is the most similar for IP distance and 0.0 is the most similar for L2

**Reply by yhmo** [reply]
I have updated the illustration. Will attach it to the document on this page: https://milvus.io/docs/single-vector-search.md#Range-search
